src/System.Net.Requests/ref/System.Net.Requests.cs

```diff
@@ -127,41 +127,62 @@ public enum WebExceptionStatus
         TrustFailure = 9,
         UnknownError = 16
     }
-    public abstract partial class WebRequest
+    public abstract partial class WebRequest : System.Runtime.Serialization.ISerializable
     {
         protected WebRequest() { }
-        public abstract string ContentType { get; set; }
+        protected WebRequest(System.Runtime.Serialization.SerializationInfo serializationInfo, System.Runtime.Serialization.StreamingContext streamingContext) { }
+        public System.Net.Security.AuthenticationLevel AuthenticationLevel { get { throw null; } set { } }
+        public virtual long ContentLength { get { throw null; } set { } }
+        public virtual string ContentType { get { throw null; } set { } }
+        public virtual System.Net.Cache.RequestCachePolicy CachePolicy { get { throw null; } set { } }
+        public virtual string ConnectionGroupName { get { throw null; } set { } }
         public virtual System.Net.ICredentials Credentials { get { return default(System.Net.ICredentials); } set { } }
+        public static System.Net.Cache.RequestCachePolicy DefaultCachePolicy { get { throw null; } set { } }
         public static System.Net.IWebProxy DefaultWebProxy { get { return default(System.Net.IWebProxy); } set { } }
-        public abstract System.Net.WebHeaderCollection Headers { get; set; }
-        public abstract string Method { get; set; }
+        public virtual System.Net.WebHeaderCollection Headers { get { throw null; } set { } }
+        public System.Security.Principal.TokenImpersonationLevel ImpersonationLevel { get { throw null; } set { } }
+        public virtual string Method { get { throw null; } set { } }
+        public virtual bool PreAuthenticate { get { throw null; } set { } }
         public virtual System.Net.IWebProxy Proxy { get { return default(System.Net.IWebProxy); } set { } }
-        public abstract System.Uri RequestUri { get; }
+        public virtual System.Uri RequestUri { get { throw null; } }
+        public virtual int Timeout { get { throw null; } set { } }
         public virtual bool UseDefaultCredentials { get { return default(bool); } set { } }
-        public abstract void Abort();
-        public abstract System.IAsyncResult BeginGetRequestStream(System.AsyncCallback callback, object state);
-        public abstract System.IAsyncResult BeginGetResponse(System.AsyncCallback callback, object state);
+        public virtual void Abort() { throw null; }
```

Comment:
> Why are these changing from abstract to virtual?

For compat (they're virtual in desktop, in Mono, etc.)  If a type tried to derive from WebRequest and didn't override these, it would succeed in desktop/Mono/etc. but fail in core.

>  And what does throw null do ?

It's how we're defining bodies of members in contracts moving forward.  We don't need the implementation to do anything, as they'll never be invoked, and ```throw null;``` is a very small amount of both code and compiled IL we can use in any non-void member.  GenAPI is now using this approach, we just haven't gone back yet and changed existing contracts to use it.